backoff.go

```diff
@@ -0,0 +1,58 @@
+package frankenphp
+
+import (
+	"sync"
+	"time"
+)
+
+type exponentialBackoff struct {
+	backoff                time.Duration
+	failureCount           int
+	mu                     sync.RWMutex
+	maxBackoff             time.Duration
+	minBackoff             time.Duration
+	maxConsecutiveFailures int
+}
+
+func newExponentialBackoff(minBackoff time.Duration, maxBackoff time.Duration, maxConsecutiveFailures int) *exponentialBackoff {
+	return &exponentialBackoff{
+		backoff:                minBackoff,
+		minBackoff:             minBackoff,
+		maxBackoff:             maxBackoff,
+		maxConsecutiveFailures: maxConsecutiveFailures,
+	}
+}
+
+// recordSuccess resets the backoff and failureCount
+func (e *exponentialBackoff) recordSuccess() {
+	e.mu.Lock()
+	defer e.mu.Unlock()
+	e.failureCount = 0
+	e.backoff = e.minBackoff
```

Comment:
```suggestion
	e.failureCount = 0
	e.backoff = e.minBackoff
	e.mu.Unlock()
```

`defer` may cost a bit (I didn't check in this specific case): https://victoriametrics.com/blog/defer-in-go/

Here the code is so simple that it's maybe not necessary to use defer?